src/components/main-view/main-view.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import axios from 'axios';
+
+import { MovieCard } from '../movie-card/movie-card';
+import { MovieView } from '../movie-view/movie-view';
+
+export class MainView extends React.Component {
+
+  constructor() {
+    super();
+
+    this.state = {
+      movies: null,
+      selectedMovie: null
+    };
+  }
+
+  componentDidMount() {
+      axios.get('https://movie-api-1684.herokuapp.com/movies')
+        .then((response) => {
+          // Assign the result to the state
+          this.setState({
+            movies: response.data,
+          });
+        })
+        .catch(function (error) {
+          console.log(error);
+        });
+    }
+
+  onMovieClick(movie) {this.setState({selectedMovie: movie});  }
+  goBack() {this.setState({selectedMovie: null});  }
+
+
+  render() {
+    const { movies, selectedMovie } = this.state;
+
+    // Before the movies have been loaded
+    if (!movies) return <div className="main-view"/>;
+
+    return (
+     <div className="main-view">
+      {selectedMovie ? <MovieView movie={selectedMovie} goBack={() => this.goBack()} />
+         : movies.map(movie => (
+           <MovieCard key={movie._id} movie={movie} onClick={movie => this.onMovieClick(movie)}/>
```

Comment:
Your api isn't sending the _id of the movie in the movies array.  This gives an error because React need it here for the key.